.github/workflows/check_unit_tests.yml

```diff
@@ -0,0 +1,50 @@
+name: unit tests check
+
+on:
+  push:
+    branches: [main]
```

Comment:
Are we certain we want to unit test only when pushing to main or should we be testing for every commit upstream? Also, im guessing these were all tested on your feature branch?